<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PageScripting</class>
 <widget class="QWidget" name="PageScripting">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1437</width>
    <height>765</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QSplitter" name="splitter_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="handleWidth">
      <number>8</number>
     </property>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="handleWidth">
       <number>8</number>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>2</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>12</number>
          </property>
          <item>
           <widget class="QPushButton" name="runButton">
            <property name="text">
             <string>Run</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Circled Play-96.png</normaloff>:/res/icons/Circled Play-96.png</iconset>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="runWindowButton">
            <property name="text">
             <string>Run in Window</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Circled Play-96.png</normaloff>:/res/icons/Circled Play-96.png</iconset>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="fullscreenButton">
            <property name="text">
             <string>Toggle Fullscreen</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/size_off.png</normaloff>:/res/icons/size_off.png</iconset>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stopButton">
            <property name="text">
             <string>Stop</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Shutdown-96.png</normaloff>:/res/icons/Shutdown-96.png</iconset>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="helpButton">
            <property name="toolTip">
             <string>Show help</string>
            </property>
            <property name="text">
             <string>Help</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Help-96.png</normaloff>:/res/icons/Help-96.png</iconset>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTabWidget" name="fileTabs">
          <property name="tabShape">
           <enum>QTabWidget::Triangular</enum>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QmlEditor" name="mainEdit">
           <attribute name="title">
            <string>main</string>
           </attribute>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QQuickWidget" name="qmlWidget">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="resizeMode">
        <enum>QQuickWidget::SizeRootObjectToView</enum>
       </property>
      </widget>
     </widget>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Triangular</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Console</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="VTextBrowser" name="debugEdit">
          <property name="font">
           <font>
            <family>DejaVu Sans Mono</family>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="clearConsoleButton">
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Delete-96.png</normaloff>:/res/icons/Delete-96.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Recent</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QListWidget" name="recentList"/>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <property name="spacing">
           <number>3</number>
          </property>
          <item>
           <widget class="QPushButton" name="openRecentButton">
            <property name="toolTip">
             <string>Open selected recent file</string>
            </property>
            <property name="text">
             <string>Open</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Open Folder-96.png</normaloff>:/res/icons/Open Folder-96.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="removeSelectedButton">
            <property name="toolTip">
             <string>Remove selected recent file from list (file will not be deleted from file system)</string>
            </property>
            <property name="text">
             <string>Remove</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Delete-96.png</normaloff>:/res/icons/Delete-96.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearRecentButton">
            <property name="toolTip">
             <string>Clear recent file list (nothing will be deleted from file system)</string>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Delete-96.png</normaloff>:/res/icons/Delete-96.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Examples</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QListWidget" name="exampleList"/>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QPushButton" name="openExampleButton">
            <property name="text">
             <string>Open</string>
            </property>
            <property name="icon">
             <iconset resource="../res.qrc">
              <normaloff>:/res/icons/Open Folder-96.png</normaloff>:/res/icons/Open Folder-96.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Tools</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Export C Array</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <property name="spacing">
              <number>2</number>
             </property>
             <item>
              <widget class="QPushButton" name="exportCArrayHwButton">
               <property name="text">
                <string>Export Array HW</string>
               </property>
               <property name="icon">
                <iconset resource="../res.qrc">
                 <normaloff>:/res/icons/Save as-96.png</normaloff>:/res/icons/Save as-96.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="exportCArrayAppButton">
               <property name="text">
                <string>Export Array App</string>
               </property>
               <property name="icon">
                <iconset resource="../res.qrc">
                 <normaloff>:/res/icons/Save as-96.png</normaloff>:/res/icons/Save as-96.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="calcSizeButton">
               <property name="toolTip">
                <string>Calculate compressed QML size</string>
               </property>
               <property name="text">
                <string>Calculate Size</string>
               </property>
               <property name="icon">
                <iconset resource="../res.qrc">
                 <normaloff>:/res/icons/Calculator-96.png</normaloff>:/res/icons/Calculator-96.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_4">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>Connected VESC</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <property name="spacing">
              <number>2</number>
             </property>
             <item>
              <widget class="QPushButton" name="openQmluiHwButton">
               <property name="text">
                <string>Open Qmlui HW</string>
               </property>
               <property name="icon">
                <iconset resource="../res.qrc">
                 <normaloff>:/res/icons/Open Folder-96.png</normaloff>:/res/icons/Open Folder-96.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="openQmluiAppButton">
               <property name="text">
                <string>Open Qmlui App</string>
               </property>
               <property name="icon">
                <iconset resource="../res.qrc">
                 <normaloff>:/res/icons/Open Folder-96.png</normaloff>:/res/icons/Open Folder-96.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_5">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>405</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>QML Upload</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <property name="spacing">
                <number>2</number>
               </property>
               <item>
                <widget class="QPushButton" name="uploadButton">
                 <property name="text">
                  <string>Erase &amp;&amp; Upload</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../res.qrc">
                   <normaloff>:/res/icons/Download-96.png</normaloff>:/res/icons/Download-96.png</iconset>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="eraseOnlyButton">
                 <property name="text">
                  <string>Erase Only</string>
                 </property>
                 <property name="icon">
                  <iconset resource="../res.qrc">
                   <normaloff>:/res/icons/Delete-96.png</normaloff>:/res/icons/Delete-96.png</iconset>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="uploadFullscreenBox">
                 <property name="toolTip">
                  <string>Run uploaded UI in fullscreen mode.</string>
                 </property>
                 <property name="text">
                  <string>Fullscreen</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QPushButton" name="clearUploadTextButton">
                 <property name="toolTip">
                  <string>Clear text below</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="icon">
                  <iconset resource="../res.qrc">
                   <normaloff>:/res/icons/Delete-96.png</normaloff>:/res/icons/Delete-96.png</iconset>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPlainTextEdit" name="uploadTextEdit">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QQuickWidget</class>
   <extends>QWidget</extends>
   <header location="global">QtQuickWidgets/QQuickWidget</header>
  </customwidget>
  <customwidget>
   <class>VTextBrowser</class>
   <extends>QTextEdit</extends>
   <header>widgets/vtextbrowser.h</header>
  </customwidget>
  <customwidget>
   <class>QmlEditor</class>
   <extends>QWidget</extends>
   <header>widgets/qmleditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>clearConsoleButton</sender>
   <signal>clicked()</signal>
   <receiver>debugEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1415</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>1128</x>
     <y>744</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearUploadTextButton</sender>
   <signal>clicked()</signal>
   <receiver>uploadTextEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1385</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>1386</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
